src/main/java/com/amplitude/api/AmplitudeClient.java

```diff
@@ -2196,6 +2211,15 @@ private String initializeDeviceId() {
             }
         }
 
+        if (useAppSetIdForDeviceId ) {
+            String appSetId = deviceInfo.getAppSetId();
+            if (!(Utils.isEmptyString(appSetId) || invalidIds.contains(appSetId))) {
+                String appSetDeviceId = appSetId + "S";
```

Comment:
Is the 'S' to mark when a device's ID is an app set ID and therefore should be retrieved again every time the SDK starts up? I don't see any documentation for the letter on Google side. Feel free to leave documentation here

Reply:
Leave a code comment is what I meant

Reply:
This is the suffix I picked to tell if in the future what we stored is app set id or not. The corresponding part is in https://github.com/amplitude/Amplitude-Android/pull/290#discussion_r687993695 So later when we init we can tell if it is app set id and we want to fetch again.

Reply:
I mean, put the comment text into our code so we remember later on. That makes sense Qingzhuo

Reply:
I see, let me add a comment for it

Reply:
add a comment for it.

Reply:
Could adid end up with S? It won't include any upper case character?

Btw, is the length fixed for app set id?

Reply:
@haoliu-amp adid now doesn't have any suffix though, and adid is UUID though, so a-f is better not to used as suffix (UUID is case inventive so A-F is also excluded). The output would be normally lowercase but just pick one they won't use for range. And app set id return value is also uuid, for example `368f80ff-404c-48d7-880e-74c5006e373f`

Reply:
@qingzhuozhen Got it.

Reply:
One thing to think about is about the attribution. In the backend, the S needs to be stripped out to enable attribution working.

Reply:
@haoliu-amp I believe they will use the `android_app_set_id` field in the event api properties to do the attribution?  That field won't have a suffix with it. But I will let them know the device id with app set id will have a suffix with it.

Reply:
I know S is to identify the source, but how can it be useful?

Reply:
@haoliu-amp The only purpose to identify the source is to re-fetch the app set id each time when we initialize, in case the app set id get updated. https://developer.android.com/training/articles/app-set-id#dont-rely-on-cached-value If we believe we don't need to re-fetch app set id for device id, then I can remove the suffix and the check to see if it is app set id. Just adding now is the best chance if we want to enable a re-fetch on app set id.

Reply:
Got it. Keep it that way.